### PR TITLE
Add possibility to include custom JS and CSS from WORKSPACE in report

### PR DESCRIPTION
**Pull request by lvogt**
This adds the cucumber-reporting feature to add custom CSS or JS files to the report.

See https://github.com/damianszczepanik/cucumber-reporting/pull/1015

### CODE REVIEW

File: src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java [hunk 1]
```diff
@@ -121,6 +125,24 @@ public void setClassifications(List<Classification> classifications) {
         }
     }
 
+    @DataBoundSetter
+    public void setCustomJsFiles(String customJsFiles) {
+        this.customJsFiles = customJsFiles;
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java:130]: Check line 123 - there is NPE prevention for backwards compatibility - don't you need same ?

File: src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java [hunk 2]
```diff
@@ -487,6 +535,15 @@ private void generateReport(Run<?, ?> build, FilePath workspace, TaskListener li
         FileUtils.deleteQuietly(directoryJsonCache);
     }
 
+    private List<String> copyFilesAndGetList(TaskListener listener, FilePath inputDirectory, File directoryJsonCache, String includePattern, String excludePattern) throws IOException, InterruptedException {
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java:538]: did you check this logic on runtime?

File: src/main/resources/net/masterthought/jenkins/CucumberReportPublisher/config.properties [hunk 3]
```diff
@@ -35,4 +35,6 @@ reducingMethods.mergeFeaturesById.title=Merge features with different JSON files
 reducingMethods.mergeFeaturesWithRetest.title=Merge features and scenarios from different JSON files of different runs into a single report by features' and scenarios' ids
 reducingMethods.hideEmptyHooks.title=Does not display hooks which do not have attachment or error message
 reducingMethods.skipEmptyJSONFiles.title=Skip empty JSON reports. If this flag is not selected then report generation fails on empty file
-presentationModes.expandAllSteps.title=Expands all steps by default
\ No newline at end of file
+presentationModes.expandAllSteps.title=Expands all steps by default
+customStyle.css=Include custom CSS files in all report pages
```
Comment (damianszczepanik) [inline src/main/resources/net/masterthought/jenkins/CucumberReportPublisher/config.properties:39]: worth to mention whether the path should be relative or absolute

Comment (lvogt) [reply src/main/resources/net/masterthought/jenkins/CucumberReportPublisher/config.properties:39]: Good point. I actually had "a bug" as I intended it to be relative to workspace, but I chose relative to the reports directory.
I fixed it and renamed a few variables to make them clearer (I hope).

File: src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java (hunk 2 above)
Comment (lvogt) [reply src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java:538]: I generated a few very simple reports, yes.
Tested as freestyle project  and two reports in one declarative pipeline.

File: src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java (hunk 1 above)
Comment (lvogt) [reply src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java:130]: I don't think so?
I only added Strings not Collections, and they are only accessed after `StringUtils.isNotEmpt()`. So it should be fine?

Comment (damianszczepanik) [reply src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java:130]: Did you run plugin without this change and then check what happens after upgrade when job is executed again after upgrade ?

Please recheck as I had such regression two times when adding new attributes

File: src/main/resources/net/masterthought/jenkins/CucumberReportPublisher/help-customCssFiles.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<p>Comma separated list of CSS files or patterns (relative to workspace) to include into all report pages.</p>
```
Comment (damianszczepanik) [inline src/main/resources/net/masterthought/jenkins/CucumberReportPublisher/help-customCssFiles.html:1]: nice

File: src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java (hunk 1 above)
Comment (lvogt) [reply src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java:130]: I was on vacation last week, but now I was finally able to test it.

What I did:
* booted a clean Jenkins instance
* installed plugin from repository
* created a freestyle job which generates a minimal report
* created report
* upgraded to "this" version of the plugin (incl. Jenkins restart)
* ran job again without any config changes (was fine)
* added custom css/js in configuration
* (successfully) ran job again with included css/js

**Comment by lvogt** [comment]
Thank you :)

**Pull request merged**